proxy/core/src/main/scala/io/cloudstate/proxy/eventing/GooglePubsubEventing.scala

```diff
@@ -0,0 +1,330 @@
+
+  /**
+   * Creates a GrpcClientSettings from this PubSubSettings
+   */
+  def createClientSettings()(implicit sys: ActorSystem): GrpcClientSettings = {
+    val sslConfig = rootCa.fold("") { rootCa =>
+      s"""
+      |ssl-config {
+      |  disabledKeyAlgorithms = []
+      |  trustManager = {
+      |    stores = [
+      |      { type = "PEM", path = "$rootCa", classpath = true }
+      |    ]
+      |  }
+      |}""".stripMargin
+    }
+
+    val akkaGrpcConfig = s"""
+      |host = "$host"
+      |port = $port
+      |
+      |$sslConfig
+      |""".stripMargin
+
+    val settings = //TODO consider using Discovery and/or other settings from: https://github.com/akka/akka-grpc/blob/master/runtime/src/main/resources/reference.conf#L36
+      GrpcClientSettings.fromConfig(
+        ConfigFactory
+          .parseString(akkaGrpcConfig)
+          .withFallback(sys.settings.config.getConfig("akka.grpc.client.\"*\""))
+      )
+
+    callCredentials
+      .fold(settings)(settings.withCallCredentials)
+      .withTls(rootCa.isDefined)
+  }
+}
+
+object GCPubsubEventingSupport {
+  final val MANUALLY = "manually"
+  final val BY_PROXY = "by-proxy"
+  final val USING_CRD = "using-crd"
+}
+
+class GCPubsubEventingSupport(config: Config, materializer: ActorMaterializer) extends EventingSupport {
+  import GCPubsubEventingSupport._
+
+  final val projectId: String = config.getString("project-id")
+  final val pollInterval: FiniteDuration = config.getDuration("poll-interval").toMillis.millis
+  final val upstreamAckDeadline: FiniteDuration =
+    config.getDuration("upstream-ack-deadline").toMillis.millis
+  final val downstreamBatchDeadline: FiniteDuration =
+    config.getDuration("downstream-batch-deadline").toMillis.millis
+  final val downstreamBatchSize: Int = config.getInt("downstream-batch-size")
+
+  private[this] final val upstreamAckDeadlineSeconds = upstreamAckDeadline.toSeconds.toInt
+
+  final val manageTopicsAndSubscriptions = config.getString("manage-topics-and-subscriptions") match {
+    case mode @ (MANUALLY | USING_CRD | BY_PROXY) => mode
+    case other =>
+      require(
+        false,
+        s"manage-topics-and-subscriptions must be one of: ${MANUALLY}, ${USING_CRD}, or ${BY_PROXY} but is '$other'"
+      )
+  }
+
+  private[this] final val settings = new PubSubSettings(config)
+
+  private def validate(): Unit = {
+    require(!projectId.isEmpty, s"project-id cannot be empty")
+    require(upstreamAckDeadline >= 10.seconds,
+            s"upstream-ack-deadline must be at least 10 seconds but was ${upstreamAckDeadline}")
+    require(upstreamAckDeadline <= 600.seconds,
+            s"upstream-ack-deadline must be at most 600 seconds but was ${upstreamAckDeadline}")
+    require(downstreamBatchSize > 0, s"downstream-batch-size must be at least 1 but was ${downstreamBatchSize}")
+  }
+
+  validate()
+
+  // Create the gRPC clients used to communicate with Google Pubsub
```

Comment:
- It may be helpful to elaborate this (already-useful_ comment to _also_ mention that while the data will be routinely coming in over an RPC channel, it should not make any material difference whether HTTP, or any (other) transport (semantic) is used as well.

Reply:
@akramtexas This file is GooglePubsubEventing.scala, which is why it specifically mentions this detail.